src/thir.md

```diff
@@ -31,6 +34,11 @@ But it has some other interesting features that distinguish it from the HIR:
   are made explicit, and method calls and overloaded operators are converted into
   plain function calls. Destruction scopes are also made explicit.
 
+- Statements, expressions, and match arms are stored separately. Statements in the `stmts` array
+  reference expressions by their index in the `exprs` array. Expressions reference subexpressions by
+  their index in the `exprs` array. Match experssions reference arms by their index in the
+  `arms` array.
```

Comment:
This should mention the `{Expr,Stmt,Arm}Id` types. Also, I think you only need to provide one example.